codec-http3/src/main/java/io/netty/handler/codec/http3/DefaultHttp3SettingsFrame.java

```diff
@@ -15,33 +15,85 @@
  */
 package io.netty.handler.codec.http3;
 
-import io.netty.util.collection.LongObjectHashMap;
-import io.netty.util.collection.LongObjectMap;
 import io.netty.util.internal.StringUtil;
+import org.jetbrains.annotations.Nullable;
 
 import java.util.Iterator;
 import java.util.Map;
 
+/**
+ * Default implementation of {@link Http3SettingsFrame}.
+ *
+ * <p>
+ * Internally backed by {@link Http3Settings}.
+ * The legacy {@link Iterable} and {@link #get(long)} / {@link #put(long, Long)} methods
+ * are preserved for backward compatibility.
+ * </p>
+ */
 public final class DefaultHttp3SettingsFrame implements Http3SettingsFrame {
 
-    private final LongObjectMap<Long> settings = new LongObjectHashMap<>(4);
+    private final Http3Settings settings;
+
+    public DefaultHttp3SettingsFrame(Http3Settings settings) {
+        this.settings = settings;
+    }
+
+    public DefaultHttp3SettingsFrame(){
+        this.settings = new Http3Settings();
```

Comment:
I wonder if we should use the default settings in this case ? 
```suggestion
        this.settings = Http3Settings defaultSettings();
```

Reply:
cant we use `new DefaultHttp3SettingsFrame(Http3Settings.defaultSettings())`